File: web/src/api/client/playback.ts
/**
 * Playback analytics API endpoints.
 */

import { apiFetch } from './base';

export interface PlaybackHeartbeatPayload {
  job_id: string;
  language: string;
  track_kind: 'original' | 'translation';
  delta_seconds: number;
}

/**
 * Send a playback heartbeat reporting listened seconds.
 * Fire-and-forget — errors are silently swallowed.
 */
export async function sendPlaybackHeartbeat(
  payload: PlaybackHeartbeatPayload,
): Promise<void> {
  try {
    await apiFetch('/api/playback/heartbeat', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload),
    });
  } catch {
    // Fire-and-forget — never block playback on analytics failures.
  }
}

File: modules/webapi/schemas/playback.py
"""Schemas for playback analytics endpoints."""

from __future__ import annotations

from pydantic import BaseModel, Field


class PlaybackHeartbeatPayload(BaseModel):
    """Client heartbeat reporting listened seconds."""

    job_id: str
    language: str
    track_kind: str  # "original" | "translation"
    delta_seconds: float = Field(ge=0, le=300)


class PlaybackHeartbeatResponse(BaseModel):
    ok: bool = True
